packages/core/test/unit/query-generator/set-constraint-checking-query.test.ts

```diff
@@ -0,0 +1,82 @@
+import { ConstraintChecking } from '@sequelize/core';
+import { expectsql, sequelize } from '../../support';
+
+const { name } = sequelize.dialect;
+const queryGenerator = sequelize.getQueryInterface().queryGenerator;
+const notSupportedError = new Error(`Deferrable constraints are not supported by ${name} dialect`);
+
+describe('QueryGenerator#setConstraintCheckingQuery', () => {
+  describe('DEFERRED constraints', () => {
+    it('generates a constraint checking query for a deferred constraint', () => {
+      expectsql(() => queryGenerator.setConstraintCheckingQuery(new ConstraintChecking.DEFERRED()), {
+        default: notSupportedError,
+        'postgres snowflake': 'SET CONSTRAINTS ALL DEFERRED',
+      });
+    });
+
+    it('generates a constraint checking query for a deferred constraint for all columns with an empty array', () => {
```

Comment:
It's less confusing to state that it's for all _constraints_ than all _columns_. I see you already fixed the misnomer in the query generator code :)

Reply:
Updated now.